rust/parquet/Cargo.toml

```diff
@@ -52,4 +53,4 @@ zstd = "0.5"
 arrow = { path = "../arrow", version = "2.0.0-SNAPSHOT" }
 
 [features]
-default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd"]
+default = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "base64"]
```

Comment:
@vertexclique `arrow` and `base64` would have to go together, should we create a feature that refers to both crates?

Reply:
Would be super nice. Meanwhile I was looking to parquet in [this PR](https://github.com/apache/arrow/pull/7873) I realized that `"snap", "brotli", "flate2", "lz4", "zstd"` part is not always needed. Since these compressions are not additive features(I am not sure, we might need to check this).

What about a feature gating like this? wdyt?:
```
default = ["core"]
core = ["arrow", "base64"]
all = ["arrow", "snap", "brotli", "flate2", "lz4", "zstd", "base64"]
```

Reply:
I assume the various compression types could depend on what the parquet source being read supports. In that case, could a user run the risk of not being able to read in a file because it supports compression? I like the `core`, `arrow` would have been more intuitive, but I suppose it clashes with the `arrow` crate

Reply:
> I assume the various compression types could depend on what the parquet source being read supports. In that case, could a user run the risk of not being able to read in a file because it supports compression?

Yes, that's true, for having less build dependencies for default set yet another idea (FWIW bikeshedding):

```
default = ["core"]
core = ["arrow", "base64"]

comp_snap = ["snap"]
comp_brotli = ["brotli"]
comp_flate = ["flate2"]
comp_lz4 = ["lz4"]
comp_zstd = ["zstd"]
```

and users add their desired compression to enable. Just tested this, it passes parquet test on master as like that (without base64).

Reply:
I think that's fine, if users expect to read parquet data which they didn't write (and thus might not know if it's compressed), they can turn on the various compression features.

Reply:
I'm not sure, removing the compression features and running tests doesn't throw any failures, so I can't see what the behaviour should be.